### PR TITLE
Promoting X-Ray consumer spans to segments

### PR DESCRIPTION
**Pull request by atshaw43**
**Description:**
Promote consumer spans to segments and follow server span naming.

**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/25177

**Testing:**
Unit Testing

### CODE REVIEW

**Review by wangzlei — APPROVED** [review]
LGTM

File: exporter/awsxrayexporter/internal/translator/segment.go [hunk 1]
```diff
@@ -172,7 +174,7 @@ func MakeSegment(span ptrace.Span, resource pcommon.Resource, indexedAttrs []str
 		}
 	}
 
-	if name == "" && span.Kind() == ptrace.SpanKindServer {
+	if name == "" && (span.Kind() == ptrace.SpanKindServer || span.Kind() == ptrace.SpanKindConsumer) {
 		// Only for a server span, we can use the resource.
```
Comment (carolabadeer) [inline exporter/awsxrayexporter/internal/translator/segment.go:178]: nit: update the comment as well to reflect the change from only using the resource service name for server spans to using it for server and consumer spans?

File: exporter/awsxrayexporter/internal/translator/segment_test.go [hunk 2]
```diff
@@ -947,8 +947,61 @@ func TestProducerSpanWithAwsRemoteServiceName(t *testing.T) {
...
+	attributes[conventions.AttributeHTTPScheme] = "https"
+	attributes[conventions.AttributeHTTPHost] = "payment.amazonaws.com"
+	attributes[conventions.AttributeHTTPTarget] = "/"
+	attributes[conventions.AttributeRPCService] = "ABC"
+
+	resource := constructDefaultResource()
+	span := constructConsumerSpan(parentSpanID, spanName, 0, "OK", attributes)
+
+	segment, _ := MakeSegment(span, resource, nil, false, nil)
+	assert.Equal(t, "signup_aggregator", *segment.Name)
```
Comment (carolabadeer) [inline exporter/awsxrayexporter/internal/translator/segment_test.go:991]: Just a clarification question - where is the `signup_aggregator` segment name coming from?

File: exporter/awsxrayexporter/internal/translator/segment_test.go [hunk 3]
```diff
@@ -947,8 +947,61 @@ func TestProducerSpanWithAwsRemoteServiceName(t *testing.T) {
 	assert.False(t, strings.Contains(jsonStr, "user"))
 }
 
+func TestConsumerSpanWithoutNameOverrides(t *testing.T) {
```
Comment (carolabadeer) [inline exporter/awsxrayexporter/internal/translator/segment_test.go:950]:
```suggestion
func TestConsumerSpanWithoutSegmentNameOverrides(t *testing.T) {
```

File: exporter/awsxrayexporter/internal/translator/segment_test.go (hunk 2 above)
Comment (atshaw43) [reply exporter/awsxrayexporter/internal/translator/segment_test.go:991]: It is coming from the resource service name.
Check out the function constructDefaultResource().

File: exporter/awsxrayexporter/internal/translator/segment_test.go (hunk 3 above)
Comment (atshaw43) [reply exporter/awsxrayexporter/internal/translator/segment_test.go]: Added

File: exporter/awsxrayexporter/internal/translator/segment.go (hunk 1 above)
Comment (atshaw43) [reply exporter/awsxrayexporter/internal/translator/segment.go]: Changed, thanks.

**Comment by atshaw43** [comment]
TODO: Have Min review the code.

File: exporter/awsxrayexporter/internal/translator/segment.go [hunk 4]
```diff
@@ -172,8 +174,8 @@ func MakeSegment(span ptrace.Span, resource pcommon.Resource, indexedAttrs []str
 		}
 	}
 
-	if name == "" && span.Kind() == ptrace.SpanKindServer {
-		// Only for a server span, we can use the resource.
+	if name == "" && (span.Kind() == ptrace.SpanKindServer || span.Kind() == ptrace.SpanKindConsumer) {
+		// Only for a server and consumer spans, we can use the resource.
 		if service, ok := resource.Attributes().Get(conventions.AttributeServiceName); ok {
```
Comment (mxiamxia) [inline exporter/awsxrayexporter/internal/translator/segment.go:179]: I want to double check if it makes sense to have consumer span (segment) name as `service.name`? If consumer span is a span with parent, then the SQS node will be merged into service node as one node.

Comment (atshaw43) [reply exporter/awsxrayexporter/internal/translator/segment.go:179]: Yes, this is the desired behavior.

**Pull request closed without merging**